src/library_dylink.js

```diff
@@ -309,10 +309,16 @@ var LibraryDylink = {
         return dynCall(sig, arguments[0], Array.prototype.slice.call(arguments, 1));
       } catch(e) {
         stackRestore(sp);
-        // Exceptions thrown from C++ exception will be integer numbers.
-        // longjmp will throw the number Infinity. Re-throw other types of
-        // exceptions using a compact and fast check.
+        // Create a try-catch guard that rethrows the Emscripten EH exception.
+#if EXCEPTION_STACK_TRACES
+        // Exceptions thrown from C++ and longjmps will be an instance of EmscriptenEH.
+        if (!(e instanceof EmscriptenEH)) throw e;
+#else
+        // Exceptions thrown from C++ will be an integer number and longjmp will throw
+        // the number Infinity. Use the compact and fast "e !== e+0" test to check if
+        // e was not a Number.
         if (e !== e+0) throw e;
+#endif
```

Comment:
I think this is included in #18665?

Reply:
PR #18665 only unifies the comment with `tools/js_manipulation.py`. So, that PR should probably land first.